crates/gpio/src/lib.rs

```diff
@@ -0,0 +1,172 @@
+use std::collections::hash_map::HashMap;
+use std::sync::Arc;
+
+use implementors::*;
+
+use slight_common::{impl_resource, BasicState};
+use slight_file::capability_store::CapabilityStore;
+use slight_file::resource::GpioResource::*;
+use slight_file::Resource;
+
+mod implementors;
+#[cfg(test)]
+mod tests;
+
+wit_bindgen_wasmtime::export!("../../wit/gpio.wit");
+wit_error_rs::impl_error!(gpio::GpioError);
+
+/// Implements the GPIO interface defined by gpio.wit.
+///
+/// This structure is responsible for constructing the pin resources described in the slightfile and providing them to the application upon request.
+///
+/// It must be [Send], [Sync], and [Clone].
+#[derive(Clone)]
+pub struct Gpio {
+    pins: HashMap<String, Result<Pin, gpio::GpioError>>,
```

Comment:
Are we sure we want to store possible errors in the `HashMap`? As a user, I'd rather fail at compile-time instead of trying to get my pin and failing at runtime.